google/cloud/pubsublite/internal/partition_publisher_test.cc

```diff
@@ -591,6 +592,45 @@ TEST_F(PartitionPublisherTest, PublishAfterShutdown) {
             (Status{StatusCode::kAborted, "`Shutdown` called"}));
 }
 
+TEST_F(PartitionPublisherTest, ShutdownWaitsForReadResponse) {
+  InSequence seq;
+
+  promise<Status> start_promise;
+  EXPECT_CALL(resumable_stream_ref_, Start)
+      .WillOnce(Return(ByMove(start_promise.get_future())));
+
+  promise<absl::optional<PublishResponse>> read_promise;
+  // first `Read` response is nullopt because resumable stream in retry loop
+  EXPECT_CALL(resumable_stream_ref_, Read)
+      .WillOnce(
+          Return(ByMove(make_ready_future(absl::optional<PublishResponse>()))))
+      .WillOnce(Return(ByMove(read_promise.get_future())));
+
+  future<Status> publisher_start_future = publisher_->Start();
+
+  auto underlying_stream = absl::make_unique<StrictMock<AsyncReaderWriter>>();
+  EXPECT_CALL(*underlying_stream,
+              Write(IsProtoEqual(GetInitializerPublishRequest()), _))
+      .WillOnce(Return(ByMove(make_ready_future(true))));
+  EXPECT_CALL(*underlying_stream, Read)
+      .WillOnce(Return(ByMove(make_ready_future(
+          absl::make_optional(GetInitializerPublishResponse())))));
+  initializer_(std::move(underlying_stream));
+
+  EXPECT_CALL(alarm_token_ref_, Destroy);
+  EXPECT_CALL(resumable_stream_ref_, Shutdown)
+      .WillOnce(Return(ByMove(make_ready_future())));
+  auto shutdown_future = publisher_->Shutdown();
+  start_promise.set_value(Status());
+  EXPECT_EQ(publisher_start_future.get(), Status());
+
+  // shutdown future does not finish until read response is received.
+  EXPECT_EQ(shutdown_future.wait_for(std::chrono::milliseconds(25)),
+            std::future_status::timeout);
```

Comment:
Hmm, any reason that `wait_for()` needs to be for 25ms?  Wouldn't any amount of time give you the same confidence?  If so, consider something shorter, like 2ms.

Reply:
I changed this to `EXPECT_FALSE(shutdown_future.is_ready());`

The wait was from an earlier iteration where there was a false positive test pass for no implementation changes. But this has now been resolved.